ee/hogai/utils/helpers.py

```diff
@@ -234,6 +243,55 @@ def extract_content_from_ai_message(response: BaseMessage) -> str:
     return str(response.content)
 
 
+def extract_thinking_from_ai_message(response: BaseMessage) -> list[dict[str, Any]]:
+    thinking: list[dict[str, Any]] = []
+
+    for content in response.content:
+        # Anthropic style reasoning
+        if isinstance(content, dict) and "type" in content:
+            if content["type"] in ("thinking", "redacted_thinking"):
+                thinking.append(content)
+    if response.additional_kwargs.get("reasoning") and (
+        summary := response.additional_kwargs["reasoning"].get("summary")
+    ):
+        # OpenAI style reasoning
+        thinking.append(
+            {
+                "type": "thinking",
+                "thinking": summary[0]["text"],
+            }
+        )
+    return thinking
+
+
+def normalize_ai_message(message: AIMessage | AIMessageChunk) -> AssistantMessage:
+    message_id = None
+    if isinstance(message, AIMessageChunk):
+        tool_calls = [
+            AssistantToolCall(
+                id=tool_call["id"],
+                name=tool_call["name"],
+                args=(tool_call["args"] if isinstance(tool_call["args"], dict) else {}),
+            )
+            for tool_call in message.tool_call_chunks
```

Comment:
It should most likely be:
```suggestion
            for tool_call in message.tool_calls
```

`tool_call_chunks` always returns args as a string.